packages/mask/src/extension/popups/modals/AddContactModal/index.tsx

```diff
@@ -0,0 +1,62 @@
+import { ActionButton, makeStyles } from '@masknet/theme'
+import { forwardRef, useState } from 'react'
+import { BottomDrawer, type BottomDrawerProps } from '../../components/index.js'
+import type { SingletonModalRefCreator } from '@masknet/shared-base'
+import { useSingletonModal } from '@masknet/shared-base-ui'
+import { useI18N } from '../../../../utils/i18n-next-ui.js'
+import { EmojiAvatar } from '@masknet/shared'
+
+const useStyles = makeStyles()((theme) => ({
+    button: {
+        flex: 1,
+    },
+    secondaryButton: {},
+    buttonGroup: {
+        marginTop: theme.spacing(2),
+        display: 'flex',
+        columnGap: 12,
+    },
+}))
+
+interface AddContactModalProps extends BottomDrawerProps {
+    onConfirm?(): void
+    address: string
+}
+
+function AddContactDrawer({ onConfirm, address, ...rest }: AddContactModalProps) {
+    const { classes, cx } = useStyles()
+    const { t } = useI18N()
+    return (
+        <BottomDrawer {...rest}>
+            <div className={classes.buttonGroup}>
+                <EmojiAvatar address={address} />
+                <ActionButton className={cx(classes.button, classes.secondaryButton)} onClick={rest.onClose}>
+                    {t('cancel')}
+                </ActionButton>
+                <ActionButton onClick={onConfirm}>{t('confirm')}</ActionButton>
+            </div>
+        </BottomDrawer>
+    )
+}
+
+export type AddContactModalOpenProps = Omit<AddContactModalProps, 'open'>
+export const AddContactModal = forwardRef<SingletonModalRefCreator<AddContactModalOpenProps, boolean>>((_, ref) => {
+    const [props, setProps] = useState<AddContactModalOpenProps>({
+        title: '',
+        address: '',
+    })
+
+    const [open, dispatch] = useSingletonModal(ref, {
```

Comment:
Should we use `<ActionModal />` here? CC @UncleBill @nuanyang233

Reply:
It can be.